src/uri.cpp

```diff
@@ -275,9 +275,7 @@ uri::query_iterator::query_iterator(optional<detail::uri_part> query)
 
 uri::query_iterator::query_iterator(const query_iterator &other)
```

Comment:
I just looked at the `query_iterator` roughly. Why does the iterator treat `;` as a separator? Couldn't find anything about that in RFC3986.

Reply:
RFC 3986 also doesn't define the '&' separator. Although '&' is most commonly used, you will sometimes still find ';' and the URI parser should work.

Reply:
Yes, you are right. One more reason that `append_query_parameter` should make use of `encode_query_component` instead of `encode_query`, because the latter does not percent-encode ';'.